Various changes for v1.1.0 release. (#496)
heremaps/here-olp-sdk-cpp: CHANGELOG.md

## v1.1.0 (11/12/2019)

**Common**

> * The deprecated `cancel` method in `CancellationToken` removed.

Reviewer: "was removed"

> * The curl network implementation fixed to compile on 32 bits architecture.

Reviewer: ```suggestion
* The `curl` network implementation was fixed and can now compile on 32 bits architecture.
```

> * The `disk_path` field in `CacheSettings` is deprecated. The `disk_path_mutable` field should be used instead.

Reviewer: ```suggestion
* The `disk_path` field in `CacheSettings` is deprecated.  Use the `disk_path_mutable` field instead.
```

> * The `disk_path` field in `CacheSettings` is deprecated. The `disk_path_mutable` field should be used instead.

Reviewer: ```suggestion
* The `disk_path` field in `CacheSettings` is deprecated. Use the `disk_path_mutable` field instead.
```

* A new synchronous `CallApi` method was added to `OlpClient`.

> * A bug is fixed when cmake was not able to check that the `pipe2` symbol exists.

Reviewer: "are enhanced"

> * A bug is fixed when cmake was not able to check that the `pipe2` symbol exists.

Reviewer 2: > "symbols detection enhanced"

Haha😁Right, keep it positive.

**olp-cpp-sdk-authentication**

> * The `SignInClient` method in `AuthenticationClient` is deprecated in favor of the newly introduced `SignInClient` method with a different signature.

Reviewer: ```suggestion
* The `SignInClient` method in `olp::authentication::AuthenticationClient` is deprecated. Use the new `SignInClient` method instead.
```

> * The `SignInClient` method in `AuthenticationClient` is deprecated in favor of the newly introduced `SignInClient` method with a different signature.

Reviewer 2: > favour 

Do we use UK English or US English?

> * A `scope` support was added through the `SignInProperties` struct.

Reviewer 3: It is not very obvious what scope support is.
Maybe "Added scope support in OAuth2 trough `olp::authentication::SignInProperties`. Users are now able to access project bound resources by using the `olp::authentication::SignInProperties::scope` field."

> * A `scope` support was added through the `SignInProperties` struct.

Reviewer: ```suggestion
* The `scope` support ws added to OAuth2 through `olp::authentication::SignInProperties`. You can now access the project bound resources using the `olp::authentication::SignInProperties::scope` field.
```

> * A new `error_id` field was added to `ErrorResponse` structure. It used to propagate `ERROR_ID` string from the authentication error response.

Reviewer 3: ```suggestion
* `error_id` field now added to `ErrorResponse` structure. It will be used to propagate the `errorId` string from the authentication error response.
```

Reviewer: ```suggestion
* The `error_id` field was added to the `ErrorResponse` structure. You can use it to get the `errorId` string from the authentication error response.
```

**olp-cpp-sdk-dataservice-read**

> * The deprecated `GetCatalogMetadataVersion` method in `CatalogClient` removed.

Reviewer 3: ```suggestion
* Removed the deprecated `GetCatalogMetadataVersion` method in `olp::dataservice::read::CatalogClient`.
```

Reviewer: ```suggestion
* The deprecated `GetCatalogMetadataVersion` method was removed from the `olp::dataservice::read::CatalogClient`.
```

**olp-cpp-sdk-dataservice-write**

> * Unused code removed.

Reviewer 3: ```suggestion
* Legacy and unused code removed.
```

Reviewer: ```suggestion
* Legacy and unused code were removed.
```
